README.md
# vCardMaker

vCardMaker is a split frontend/backend application with PHP and C# backends for creating version 2.1 and 3.0 vCards.

The C# version can be used both with Windows/IIS and Linux running Mono. The PHP version also includes a frontend.

The `_zip` folder contains original zipped source code for all versions of the application.

## Setup

As well as customising any strings, remember to include a valid email address at the bottom of your chosen script (vCardMaker.php or vCardMaker.cs) where denoted. This email address will appear as the "from" address in any emails sent by the backend scripts.


CHANGELOG.TXT
vCardMaker Changelog
====================

1.1.0 (Bring Me To Life)
------------------------
 - Relicensed under GNU GPL 3 or later
 - Removed some implementation-specific data

1.0.1 (Me and My Imagination)
-----------------------------
Frontend
 - Redesigned the homepage to show fewer form fields by default
 - Updated copyright dates

1.0.0 (Encore Une Fois)
-----------------------
Frontend
 - Amended fields to match names changes in backend
Backend
 - Changed the following field names for consistency
  - Version.vCard -> vCard.Version
  - Name -> vCard.Name
  - Source -> vCard.Source
  - Formatted.Name -> Name.Formatted
  - Name.AdditionalNames -> Name.Additional
  - Birthdate -> Birth.Date
  - Birthtime -> Birth.Time
  - Birthtimezone -> Birth.Time.Zone
  - Photo.Source.Quoted and Photo.Source.URL -> Photo.Source
  - Logo.Source.Quoted and Logo.Source.URL -> Logo.Source
  - Agent and Agent.URL -> Agent.Source
  - Sound.Source.Phonetic and Sound.Source.URL and Sound.Source.Quoted -> Sound.Source
  - Key.Quoted -> Key.Source
  - Comment.Quoted -> Comment
  - GeographicalProperties.Position.* -> GeographicalProperties.Position.Latitude and GeographicalProperties.Position.Longitude

0.8.2 (Play Me)
---------------
Backend
 - Fixed bugs where blank values for photo, logo, agent, sound or key fields would generate stub entries in the vCard

0.8.1 (Imaginary)
-----------------
Backend
 - Added function and changed others to facilitate future API calls

0.8.0 (After All)
-----------------
Frontend
 - Changed wizard style to allow users to see all pages at once
 - Removed the progress bar
 - Fixed bug where help page would open on second monitor on multi-monitor setups
 - Updated icons
 - Changed handling of results
 - Changed to IE7 emulation to mitigate IE8 beta bugs
  - Resized select box arrow is cut off
  - Showing of results slows down after few clicks
 - Upgraded to latest version of jQuery, now hosted on Google
 - Fixed bug where selecting an autocomplete suggestion would not correctly update the formatted name field
 - Restored parity between the PHP and ASP.NET/C# versions

0.7.2 (On My Mind)
------------------
Frontend
 - Removed all advertisements
 - Reformatted and re-styled the form elements
 - Generally cleaned up the interface

0.7.1 (It's Over Now)
---------------------
Backend
 - Fixed a few small source code formatting issues
 - Mailer name is now trimmed
 - Updated the country list (correct as of 15 May 2008)
 - Fixed bug where results were not being shown in Internet Explorer

0.7.0 (Pink Love)
-----------------
Frontend
 - Changed to a wizard interface
 - Added a timeline
 - Moved to jQuery 1.2.1
 - Moved optional and advanced fields to new advanced options sections
 - Added a calendar for birth date
Backend
 - Released vCardMaker PHP backend on Sourceforge.net
 - Updated the country list (correct as of 17 January 2008)

0.6.4
-----
 - Changed all inline onclick event handlers to onchange handlers
 - Added an Google AdSense advertisement to the bottom of the page

0.6.3
-----
 - Added instructions to the top of the page
 - Fixed all outstanding PHP errors
 - Added ability to email vCards
 - Made at least one email address compulsory
 - Fixed bug where blank vCard version would spew out errors
 - Disabled certain options for the primary email
 - Combined vCard version and "name and source"

0.6.2
-----
 - Tweaked some styles
 - Removed the donation button
 - Made the footer smaller and updated it
 - Changed some text to be clearer
 - Removed all co-branding features
 - Updated the country list (correct as of 27 May 2007)
 - Results now appear in an overlay
 - vCardMaker now works with JavaScript disabled
 - Moved to mootools 1.1
 - Changed page title
 - Made sections collapsible
 - Made text boxes larger

0.6.1
-----
 - Added mandatory TYPE parameter list prefix for 3.0 vCards (user request)
   - Address
   - Address Label
   - Telephone number
   - Email address
 - Made URL the default for sounds rather than phonetic
 - Fixed address label quoted-printable encoding

0.6.0
-----
 - Make a lot of changes to the code to facilitate co-branding
 - PNGCRUSHed the title image
 - Tweaked the vCardMaker UI
 - Added contact link
 - Removed "Report a bug" link
 - Fixed a bug where the formatted name was not reset with the rest of the form
 - Removed contextual help - help will be back later in another form
 - Removed the version number in the footer
 - Removed lots of unused JavaScript libraries
 - Removed Behaviour in favour of plain JavaScript
 - vCard begin and end statements are no longer automatically added to agent vCards

0.5.3
-----
 - Added birth time and timezone
 - Made many more text strings translatable
 - Added agent URL for 3.0 vCards
 - Added print styles
 - Updated the country list (correct as of 15 June 2006)

0.5.2
-----
 - Added PayPal donation button
 - Removed onkeypress attributes from links
 - Amended some help text
 - Amended some styles

0.5.1
-----
 - Added escaping of colons, semicolons and commas in all fields
 - Added preferred email address checkbox

0.5.0
-----
 - Added moo.ajax and Behaviour to the JavaScript library
 - Completed the switch to a completely AJAX solution
 - Started the move to unobtrusive JavaScript
 - Added form reset facility

0.4.2
-----
 - Cleaned up the interface a little
 - Fixed timezones
 - Changed contextual help to JavaScript windows
 - Fixed a couple of JavaScript problems

0.4.1
-----
 - Added contextual help to every section

0.4.0
-----
 - Reformatted the vCardMaker form (removed all tables)
 - Moved some interface elements around
 - Moved last modified date and UID generation to vCard class
 - Added changelog link to footer
 - Changed photo, logo, sound and key choosers from radio boxes to select boxes

PHP/Make.php
<?php

/*
    This file is part of vCardMaker.

    vCardMaker is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    vCardMaker is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with vCardMaker.  If not, see <http://www.gnu.org/licenses/>.
*/

require_once 'vCardMaker.php';
$vCardMaker = new vCardMaker();
echo $vCardMaker->Make($_POST);

?>

PHP/vCardMaker_Lang.php
<?php

/*
    This file is part of vCardMaker.

    vCardMaker is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    vCardMaker is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with vCardMaker.  If not, see <http://www.gnu.org/licenses/>.
*/

$vCardMaker_Lang = array();

// Validator strings
$vCardMaker_Lang['NoVersion']            = 'The vCard version is blank.';
$vCardMaker_Lang['NoFormattedName']      = 'The full name is blank.';
$vCardMaker_Lang['NoForename']           = 'The forename is blank.';
$vCardMaker_Lang['NoSurname']            = 'The surname is blank.';
$vCardMaker_Lang['NoNameSource']         = 'If name or source are used, then both must be provided.';
$vCardMaker_Lang['NoBirthDate']          = 'The birth date is not in the correct format.';
$vCardMaker_Lang['NoBirthDateTime']      = 'If the birth time is provided then the birth date must be provided.';
$vCardMaker_Lang['NoBirthTime']          = 'The birth time is not in the correct format.';
$vCardMaker_Lang['NoBirthTimeZone']      = 'If the birth timezone is provided then the birth time must be provided.';
$vCardMaker_Lang['NoSoundPhonetic']      = 'The phonetic type for a sound cannot be used in a 3.0 vCard.';
$vCardMaker_Lang['NoSoundSource']        = 'The sound source is blank.';
$vCardMaker_Lang['NoPhotoSource']        = 'The photo source is blank.';
$vCardMaker_Lang['NoPhotoQP']            = 'The Quoted-Printable type for a photo cannot be used in a 3.0 vCard.';
$vCardMaker_Lang['NoDeliveryStreet']     = 'The street name for your address is blank.';
$vCardMaker_Lang['NoDeliveryLocality']   = 'The locality for your address is blank.';
$vCardMaker_Lang['NoDeliveryPostCode']   = 'The postal code for your address is blank.';
$vCardMaker_Lang['NoDeliveryCountry']    = 'The country for your address is blank.';
$vCardMaker_Lang['NoLabelStreet']        = 'The street name for your delivery label is blank.';
$vCardMaker_Lang['NoLabelLocality']      = 'The locality for your delivery label is blank.';
$vCardMaker_Lang['NoLabelPostCode']      = 'The postal code for your delivery label is blank.';
$vCardMaker_Lang['NoLabelCountry']       = 'The country for your delivery label is blank.';
$vCardMaker_Lang['NoTelNumber']          = 'The telephone number is blank.';
$vCardMaker_Lang['NoEmailAddress']       = 'The email address is blank.';
$vCardMaker_Lang['NoGeoLoc']             = 'The geographical location is incorrect.';
$vCardMaker_Lang['NoLogoSource']         = 'The logo source is blank.';
$vCardMaker_Lang['NoLogoQP']             = 'The Quoted-Printable type for a logo cannot be used in a 3.0 vCard.';
$vCardMaker_Lang['NoAgentSource']        = 'The agent source is blank.';
$vCardMaker_Lang['NoAgentURLType']       = 'The URL agent type cannot be used in a 2.1 vCard.';
$vCardMaker_Lang['NoKeyBinary']          = 'The binary key type cannot be used in a 2.1 vCard.';
$vCardMaker_Lang['NoKeyPGPX509']         = 'The PGP and X.509 key types cannot be used in a 3.0 vCard.';
$vCardMaker_Lang['NoKeySource']          = 'The key source is blank.';
$vCardMaker_Lang['VErrorsFound']         = 'The following errors were found:';

// Email strings
$vCardMaker_Lang['ESubject']             = 'Your vCard from vCardMaker';
$vCardMaker_Lang['EMessage']             = "Hello!\n\nYou recently created a vCard using vCardMaker and asked us to email it to you. Your vCard is included below between the dashes. If you didn't request this message, please reply to this email and we'll do our best to make sure it doesn't happen again.\n\nThank you for using vCardMaker!";

?>

PHP/Help.php
<?php

/*
    This file is part of vCardMaker.

    vCardMaker is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    vCardMaker is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with vCardMaker.  If not, see <http://www.gnu.org/licenses/>.
*/

?>
<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<title>vCardMaker Help</title>
<link rel="stylesheet" type="text/css" href="styles.css" media="all">
<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.4.0/jquery.min.js"></script>
<script type="text/javascript">
// <![CDATA[
// Generic show and hide toggle element
var state;
function vCardShowHideToggle(element, showhide) {
  if (showhide == 'in') {
    state = 'up';
    document.getElementById(element+'toggleicon').src = 'images/up.png';
    $('#'+element).slideDown('slow');
  } else if (showhide == 'out') {
    state = 'down';
    document.getElementById(element+'toggleicon').src = 'images/down.png';
    $('#'+element).slideUp('slow');
  } else {
    if (state == 'down') {
      state = 'up';
      document.getElementById(element+'toggleicon').src = 'images/up.png';
    } else {
      state = 'down';
      document.getElementById(element+'toggleicon').src = 'images/down.png';
    }
    $('#'+element).slideToggle('slow');
  }
}

$(document).ready(function() {
  // Hide the more info section
  vCardShowHideToggle('moreinfo', 'out');
  // Add toggle behaviour to the more info section
  $('a#moreinfotoggle').click(function() {
    vCardShowHideToggle('moreinfo');
    return false;
  });
  // Open certain links in a new window
  $('a._new_window').click(function() {
    window.open(this.href, '', '');
    return false;
  });
});
// ]]>
</script>
</head>

<body class="help">

<div id="staticcontent">

<h2>vCardMaker Help</h2>

<?php
    
    $isValid = false;

    switch ($_SERVER['QUERY_STRING'])
    {
        case 'Forename':
            echo '<div class="versioninfo">This field is required.</div><p>Enter your given name.</p>';
            $isValid = true;
            break;
        case 'Surname':
            echo '<div class="versioninfo">This field is required</div><p>Enter your family name.</p>';
            $isValid = true;
            break;
        case 'Photo':
            echo '<div class="versioninfo"><span class="vcard-only21">Some options in this field are only available in version 2.1 vCards.</span></div><p>Select either <strong>URL</strong> to provide an address to a photo or <strong>Base64</strong>/<strong class="vcard-only21">Quoted-Printable</strong> to enter an inline image. For the latter you must also select the type of photo from the list.</p>';
            $isValid = true;
            break;
        case 'BirthDate':
            echo '<p>Enter your date of birth, using the calendar if required. The correct format is <strong>YYYY-MM-DD</strong>.</p>';
            $isValid = true;
            break;
        case 'Version':
            echo '<div class="versioninfo">This field is required.</div><p>Select either <strong class="vcard-only21">2.1</strong> or <strong class="vcard-only30">3.0</strong> depending on the vCard version you require. If you\'re not sure, select <strong class="vcard-only21">2.1</strong>.</p>';
            $isValid = true;
            break;
        case 'vCardName':
            echo '<div class="versioninfo"><span class="vcard-only30">This field is only available in version 3.0 vCards.</span></div><p>Enter a name for this vCard. This is useful if you have multiple vCards.</p>';
            $isValid = true;
            break;
        case 'vCardSourceURL':
            echo '<div class="versioninfo"><span class="vcard-only30">This field is only available in version 3.0 vCards.</span></div><p>Enter the URL where a copy of the vCard may be found and downloaded.</p>';
            $isValid = true;
            break;
        case 'FullName':
            echo '<div class="versioninfo">This field is required.</div><p>Enter your full name as you would like it to be displayed including any titles, middle names and suffixes. <strong>This field is filled out automatically when you enter your forename, surname, middles names, title and suffix.</strong></p>';
            $isValid = true;
            break;
        case 'Title':
            echo '<p>Enter your title (such as Mr., Sir etc.)</p>';
            $isValid = true;
            break;
        case 'MiddleNames':
            echo '<p>Enter your middle name(s).</p>';
            $isValid = true;
            break;
        case 'Suffix':
            echo '<p>Enter any suffixes to your name (such as organisational memberships, qualifications etc.)</p>';
            $isValid = true;
            break;
        case 'SortOrder':
            echo '<div class="versioninfo"><span class="vcard-only30">This field is only available in version 3.0 vCards.</span></div><p>Enter your forename or surname to specify where your vCard should appear when sorted alphabetically alongside other vCards.</p>';
            $isValid = true;
            break;
        case 'Nicknames':
            echo '<div class="versioninfo"><span class="vcard-only30">This field is only available in version 3.0 vCards.</span></div><p>Enter any nickname(s) by which you are known.</p>';
            $isValid = true;
            break;
        case 'BirthTime':
            echo '<div class="versioninfo"><span class="vcard-only30">This field is only available in version 3.0 vCards.</span></div><p>Enter your time of birth. The correct format is <strong>HH:MM:SS</strong>. Seconds may be left out.</p>';
            $isValid = true;
            break;
        case 'BirthTimezone':
            echo '<div class="versioninfo"><span class="vcard-only30">This field is only available in version 3.0 vCards.</span></div><p>Select the time zone of your time of birth.</p>';
            $isValid = true;
            break;
        case 'DeliveryAddressType':
            echo '<p>Select the type of address you will be entering below. You may choose one or more types.</p>';
            $isValid = true;
            break;
        case 'DeliveryAddressPOAddress':
            echo '<p>Enter the PO Box details of this address.</p>';
            $isValid = true;
            break;
        case 'DeliveryAddressBuildingName':
            echo '<p>Enter the building name or number of this address.</p>';
            $isValid = true;
            break;
        case 'DeliveryAddressStreet':
            echo '<p>Enter the street name of this address.</p>';
            $isValid = true;
            break;
        case 'DeliveryAddressLocality':
            echo '<p>Enter the town/city of this address.</p>';
            $isValid = true;
            break;
        case 'DeliveryAddressRegion':
            echo '<p>Enter the county/state of this address.</p>';
            $isValid = true;
            break;
        case 'DeliveryAddressPostalCode':
            echo '<p>Enter the postal code of this address.</p>';
            $isValid = true;
            break;
        case 'DeliveryAddressCountry':
            echo '<p>Select the country of this address.</p>';
            $isValid = true;
            break;
        case 'DeliveryLabelType':
            echo '<p>Select the type of address you will be entering below. You may choose one or more types.</p>';
            $isValid = true;
            break;
        case 'DeliveryLabelPOAddress':
            echo '<p>Enter the PO Box details of this address.</p>';
            $isValid = true;
            break;
        case 'DeliveryLabelBuildingName':
            echo '<p>Enter the building name or number of this address.</p>';
            $isValid = true;
            break;
        case 'DeliveryLabelStreet':
            echo '<p>Enter the street name of this address.</p>';
            $isValid = true;
            break;
        case 'DeliveryLabelLocality':
            echo '<p>Enter the town/city of this address.</p>';
            $isValid = true;
            break;
        case 'DeliveryLabelRegion':
            echo '<p>Enter the county/state of this address.</p>';
            $isValid = true;
            break;
        case 'DeliveryLabelPostalCode':
            echo '<p>Enter the postal code of this address.</p>';
            $isValid = true;
            break;
        case 'DeliveryLabelCountry':
            echo '<p>Select the country of this address.</p>';
            $isValid = true;
            break;
        case 'TelephoneNumberType':
            echo '<p>Select the type of telephone number you will be entering below.</p>';
            $isValid = true;
            break;
        case 'TelephoneNumberPreferred':
            echo '<p>Select this option to show that this telephone number is your preferred one. This option is automatically selected for the first telephone number and cannot be selected for any other telephone number.</p>';
            $isValid = true;
            break;
        case 'TelephoneNumberMessaging':
            echo '<p>Select this option to show that there is a messaging service (e.g. answering machine or voicemail) active on this telephone number.</p>';
            $isValid = true;
            break;
        case 'TelephoneNumber':
            echo '<p>Enter your telephone number.</p>';
            $isValid = true;
            break;
        case 'EmailAddressType':
            echo '<p>Select the type of email address you will be entering below. The type of the first email address is always "Internet SMTP".</p>';
            $isValid = true;
            break;
        case 'EmailAddressPreferred':
            echo '<div class="versioninfo"><span class="vcard-only30">This field is only available in version 3.0 vCards.</span></div><p>Select this option to show that this email address is your preferred one. This option is automatically selected for the first email address and cannot be selected for any other email address.</p>';
            $isValid = true;
            break;
        case 'EmailAddress':
            echo '<div class="versioninfo">The first instance of this field is required.</div><p>Enter your email address.</p>';
            $isValid = true;
            break;
        case 'Mailer':
            echo '<p>Enter the name of the application/service you mainly use to send email.</p>';
            $isValid = true;
            break;
        case 'OrganisationName':
            echo '<p>Enter the name of the organisation you work for.</p>';
            $isValid = true;
            break;
        case 'JobTitle':
            echo '<p>Enter your job title.</p>';
            $isValid = true;
            break;
        case 'OrganisationalUnit':
            echo '<p>Enter the name of the unit (i.e. department) you work in within your organisation.</p>';
            $isValid = true;
            break;
        case 'JobRole':
            echo '<p>Enter a slightly expanded version of your job title.</p>';
            $isValid = true;
            break;
        case 'OrganisationLogo':
            echo '<div class="versioninfo"><span class="vcard-only21">Some options in this field are only available in version 2.1 vCards.</span></div><p>Select either <strong>URL</strong> to provide an address to a logo or <strong>Base64</strong>/<strong class="vcard-only21">Quoted-Printable</strong> to enter an inline logo. For the latter you must also select the type of logo from the list.</p>';
            $isValid = true;
            break;
        case 'Agent':
            echo '<div class="versioninfo"><span class="vcard-only30">Some options in this field are only available in version 3.0 vCards.</span></div><p>Select either <strong class="vcard-only30">URL</strong> to provide an address to the vCard of a person who should be contacted for enquiries in your absence or <strong>vCard</strong> to include their vCard directly in yours.</p>';
            $isValid = true;
            break;
        case 'URL':
            echo '<p>Enter your personal website address or that of your employer.</p>';
            $isValid = true;
            break;
        case 'Timezone':
            echo '<p>Select the time zone of your address.</p><p>For help with working out your time zone, see the <a href="http://en.wikipedia.org/wiki/List_of_time_zones" class="_new_window">list of time zones</a> along with principal cities.</p>';
            $isValid = true;
            break;
        case 'AddressPosition':
            echo '<p>Enter the geographical location of your address.</p>';
            $isValid = true;
            break;
        case 'Categories':
            echo '<div class="versioninfo"><span class="vcard-only30">This field is only available in version 3.0 vCards.</span></div><p>Enter a comma-separated list of categories that this vCard fits into (e.g. work, home etc.)</p>';
            $isValid = true;
            break;
        case 'Comments':
            echo '<p>Enter any additional comments to be attached to this vCard.</p>';
            $isValid = true;
            break;
        case 'Sound':
            echo '<div class="versioninfo"><span class="vcard-only21">Some options in this field are only available in version 2.1 vCards.</span></div><p>Select either <strong class="vcard-only21">Phonetic</strong> to enter a phonetic representation of your full name, <strong>URL</strong> to provide an address to a sound file of the pronunciation or <strong>Base64</strong> to enter an inline sound file.</p>';
            $isValid = true;
            break;
        case 'Class':
            echo '<div class="versioninfo"><span class="vcard-only30">This field is only available in version 3.0 vCards.</span></div><p>Select a classification for your vCard. <strong>Note that this is purely informational and DOES NOT provide any actual protection for your vCard.</strong></p>';
            $isValid = true;
            break;
        case 'EncryptionKey':
            echo '<div class="versioninfo"><span class="vcard-only30">Some options in this field are only available in <span class="vcard-only21">version 2.1</span> or version 3.0 vCards.</span></div><p>Select a key type (either <strong class="vcard-only21">PGP</strong>, <strong class="vcard-only21">X.509</strong> or <strong class="vcard-only30">Binary</strong> and enter your public key. <strong>Do not enter your private key here.</strong></p>';
            $isValid = true;
            break;
        default:
            echo '<div class="versioninfo">This help topic cannot be found.</div>';
            break;
    }

    if ($isValid)
    {

?>

<a href="#" id="moreinfotoggle"><img src="images/down.png" width="25" height="30" alt="" style="vertical-align: middle;" id="moreinfotoggleicon"> More information on alerts</a>
<div id="moreinfo">
<p>Fields or options coloured <strong class="vcard-only21">blue</strong> are only available when creating version 2.1 vCards while <strong class="vcard-only30">red</strong> ones are only available when creating version 3.0 vCards. All other options are universally available.</p>
<p>Possible alerts are:</p>
<ul>
<li><strong>This field is required</strong>: The field that this topic refers to is compulsory for your vCard.</li>
<li><strong>The first instance of this field is required</strong>: Where there is more than one instance of a particular field (such as email address), then one is compulsory and the rest are optional.</li>
<li><strong>This field is only available in version <em>version number</em> vCards</strong>: You can only enter data in this field if you have chosen to make a <em>version number</em> vCard.</li>
<li><strong>Some options in this field are only available in version <em>version number</em> [or version <em>version number</em>] vCards</strong>: If you have chosen certain options for this field (coloured blue or red), then you can only enter data if you have chosen to make a <em>version number</em> vCard. In this case, options which require a specific vCard version are coloured either blue or red in the help topic.</li>
</ul>
</div>

<?php
    }
?>

</div>

</body>
</html>